File: SurfaceViewDemo/app/src/main/java/edu/cs4730/surfaceviewdemo/MainActivity.java
package edu.cs4730.surfaceviewdemo;

import android.content.Intent;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;

import edu.cs4730.surfaceviewdemo.databinding.ActivityMainBinding;


public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        binding.button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getBaseContext(), AllinOneActivity.class)
                );
            }
        });

        binding.button2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(getBaseContext(), SeparateActivity.class)
                );
            }
        });
    }

}

File: TextureViewDemo_kt/app/src/main/java/edu/cs4730/textureviewdemo_kt/MainActivity.kt
package edu.cs4730.textureviewdemo_kt

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<View>(R.id.button).setOnClickListener {
            startActivity(Intent(baseContext, AllinOneActivity::class.java))
        }

        findViewById<View>(R.id.button2).setOnClickListener {
            startActivity(Intent(baseContext, SeparateActivity::class.java))
        }
    }
}